msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -309,12 +309,17 @@ class AccountState private constructor(
                             correlationId = correlationId
                         )
 
-                val privateCorrelationId = if (correlationId == "UNSET") { UUID.randomUUID().toString() } else { correlationId }
+                try {
+                    UUID.fromString(correlationId)
```

Comment:
nit: maybe assign the result to a variable and use it in the builder to avoid calling the fromString() method twice

Reply:
@Yuki-YuXin  let's run the tests without this fix here to ensure it works not. But let's keep this fix here as well, to avoid any production crashes (but clean it up a bit, e.g. call `DiagnosticContext.INSTANCE.getThreadCorrelationId()` so that a new UUID is generated in there, rather than here at the call site).

Reply:
Local GetAccessTokenTests passed with and without the changes here.